### ISSUE TITLE
CMakePresets: Preset not shown when a common dependency is inherited more than once

### ISSUE THREAD

**Issue by dsterk-x** [microsoft/vscode-cmake-tools#2210]
### Brief Issue Summary

It seems presets are not shown in the list of available presets if a common dependency is inherited more than once. For example via a transitive inheritance path.

#### How to reproduce

Using the example CMakePrests.json below. Then do Ctrl+Alt+P then type CMake: Select Configure Preset 

``` json
{
    "version": 2,
    "configurePresets": [
        {
            "name": "default",
            "displayName": "default",
            "generator": "Ninja",
            "binaryDir": "${sourceDir}/build/default"
        },
        {
            "name": "foo",
            "inherits": "default",
            "displayName": "foo"
        },
        {
            "name": "foobar",
            "inherits": [
                "foo",
                "default"
            ],
            "displayName": "foobar"
        }
    ]
}
```

#### Expected Behavior

The above mentioned procedure shows the configure presets: "default", "foo" and "foobar".

#### Actual Behavior

The above mentioned procedure shows the configure presets: "default" and "foo". "foobar" is not listed.
When calling "cmake --list-presets", the output shows all expected presets.

``` bash
$ cmake --list-presets
Available configure presets:

  "default" - default
  "foo"     - foo
  "foobar"  - foobar
``` 

It seems this behavior is related to inheritance of presets. If "foo" would not inherit from "default" all expected presets would be listed.

### CMake Tools Diagnostics

```json
{
  "os": "win32",
  "vscodeVersion": "1.61.2",
  "cmtVersion": "1.9.1",
  "configurations": [
    {
      "folder": "d:\\repos\\defective_preset",
      "cmakeVersion": "3.20.1",
      "configured": false,
      "generator": "",
      "usesPresets": true,
      "compilers": {}
    }
  ],
  "cpptoolsIntegration": {
    "isReady": false,
    "hasCodeModel": false,
    "activeBuildType": "",
    "buildTypesSeen": [],
    "requests": [],
    "responses": [],
    "partialMatches": [],
    "targetCount": 0,
    "executablesCount": 0,
    "librariesCount": 0,
    "targets": []
  },
  "settings": [
    {
      "communicationMode": "automatic",
      "useCMakePresets": "auto",
      "configureOnOpen": false
    }
  ]
}

```

**Comment by bobbrow** [comment]
We plan to release the fix for this on Monday. If you would like to verify the fix in a pre-release version of the extension, please do the following:
1. Download the VSIX here: https://github.com/microsoft/vscode-cmake-tools/suites/4455203062/artifacts/118295075
2. Unzip the VSIX file from the archive
3. Invoke the `Install from VSIX...` command in VS Code and select the VSIX file from step 2.

**Comment by dsterk-x** [comment]
Thanks for the hint. I tried the fix and for me it works again as expected.

**Comment by bobbrow** [comment]
The fix for this has been released in 1.9.2.
